File: dashpilot-android/app/src/main/java/com/softwiredtech/dashpilot/ui/onboarding/OnboardingPageScaffold.kt
package com.softwiredtech.dashpilot.ui.onboarding

import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.softwiredtech.dashpilot.ui.theme.OnboardingColors

/**
 * Shared vertical layout: hero takes the upper flex region, title/subtitle anchor
 * mid-screen, optional extra content goes between subtitle and CTA, primary CTA pins bottom.
 */
@Composable
fun OnboardingPageScaffold(
    title: String,
    subtitle: String,
    cta: @Composable () -> Unit,
    hero: (@Composable () -> Unit)? = null,
    extra: (@Composable () -> Unit)? = null,
) {
    Column(
        modifier = Modifier.fillMaxSize(),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Box(
            modifier = Modifier.weight(1f).fillMaxWidth(),
            contentAlignment = Alignment.Center
        ) {
            hero?.invoke()
        }
        Text(
            text = title,
            color = OnboardingColors.TextPrimary,
            fontSize = OnboardingTokens.PageTitle,
            fontWeight = FontWeight.Medium,
            textAlign = TextAlign.Center,
            modifier = Modifier.padding(horizontal = 16.dp)
        )
        Spacer(Modifier.height(8.dp))
        Text(
            text = subtitle,
            color = OnboardingColors.TextSecondary,
            fontSize = OnboardingTokens.Body,
            textAlign = TextAlign.Center,
            lineHeight = (OnboardingTokens.Body.value * 1.5f).sp,
            modifier = Modifier.padding(horizontal = 16.dp)
        )
        if (extra != null) {
            Spacer(Modifier.height(20.dp))
            extra()
        }
        Spacer(Modifier.height(24.dp))
        cta()
        Spacer(Modifier.height(16.dp))
    }
}

@Composable
fun PrimaryCta(
    label: String,
    onClick: () -> Unit,
    icon: ImageVector? = null,
    showSpinner: Boolean = false,
    enabled: Boolean = true,
) {
    Button(
        onClick = onClick,
        enabled = enabled,
        shape = RoundedCornerShape(OnboardingTokens.RadiusButton),
        colors = ButtonDefaults.buttonColors(
            containerColor = OnboardingColors.Accent,
            contentColor = Color.Black,
            disabledContainerColor = OnboardingColors.Accent,
            disabledContentColor = Color.Black,
        ),
        contentPadding = PaddingValues(vertical = 14.dp, horizontal = 24.dp),
        modifier = Modifier.fillMaxWidth(),
    ) {
        when {
            showSpinner -> {
                CircularProgressIndicator(
                    color = Color.Black,
                    strokeWidth = 2.dp,
                    modifier = Modifier.size(18.dp),
                )
                Spacer(Modifier.size(10.dp))
            }
            icon != null -> {
                Icon(icon, contentDescription = null, tint = Color.Black, modifier = Modifier.size(18.dp))
                Spacer(Modifier.size(10.dp))
            }
        }
        Text(label, fontSize = 14.sp, fontWeight = FontWeight.Medium)
    }
}

File: dashpilot-ios/dashpilot/UI/Onboarding/OnboardingPageScaffold.swift
import SwiftUI

/// Shared visual tokens for the onboarding flow (port of the Android
/// `OnboardingColors` / `OnboardingTokens`).
enum OnboardingStyle {
    static let accent = Color(red: 0x22 / 255.0, green: 0xC5 / 255.0, blue: 0x5E / 255.0)
    static let background = Color.black
    static let surface = Color.white.opacity(0.08)
    static let textPrimary = Color.white
    static let textSecondary = Color.white.opacity(0.55)
    static let textMuted = Color.white.opacity(0.45)
    static let stepInactive = Color.white.opacity(0.18)
    static let ledDim = Color(red: 0x1F / 255.0, green: 0x3A / 255.0, blue: 0x1F / 255.0)

    static let radiusButton: CGFloat = 12
    static let radiusSmall: CGFloat = 8

    static let topTitleSize: CGFloat = 22
    static let pageTitleSize: CGFloat = 18
    static let bodySize: CGFloat = 13
    static let captionSize: CGFloat = 11
}

/// Shared vertical layout for onboarding pages (port of the Android
/// `OnboardingPageScaffold`): hero takes the upper flex region, title/subtitle
/// anchor mid-screen, optional extra content sits between subtitle and CTA,
/// and the primary CTA pins to the bottom.
struct OnboardingPageScaffold<Hero: View, Extra: View, Cta: View>: View {
    let title: String
    let subtitle: String
    @ViewBuilder let hero: Hero
    @ViewBuilder let extra: Extra
    @ViewBuilder let cta: Cta

    var body: some View {
        VStack(spacing: 0) {
            ZStack {
                hero
            }
            .frame(maxWidth: .infinity, maxHeight: .infinity)

            Text(title)
                .foregroundColor(OnboardingStyle.textPrimary)
                .font(.system(size: OnboardingStyle.pageTitleSize, weight: .medium))
                .multilineTextAlignment(.center)
                .padding(.horizontal, 16)

            Spacer().frame(height: 8)

            Text(subtitle)
                .foregroundColor(OnboardingStyle.textSecondary)
                .font(.system(size: OnboardingStyle.bodySize))
                .multilineTextAlignment(.center)
                .lineSpacing(OnboardingStyle.bodySize * 0.5)
                .padding(.horizontal, 16)

            if !(extra is EmptyView) {
                Spacer().frame(height: 20)
                extra
            }

            Spacer().frame(height: 24)
            cta
            Spacer().frame(height: 16)
        }
    }
}

/// Full-width accent CTA button (port of the Android `PrimaryCta`).
struct OnboardingPrimaryButton: View {
    let label: String
    var systemImage: String? = nil
    var showSpinner = false
    var enabled = true
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            HStack(spacing: 10) {
                if showSpinner {
                    ProgressView()
                        .controlSize(.small)
                        .tint(.black)
                } else if let systemImage {
                    Image(systemName: systemImage)
                        .font(.system(size: 15, weight: .medium))
                }
                Text(label)
                    .font(.system(size: 14, weight: .medium))
            }
            .foregroundColor(.black)
            .frame(maxWidth: .infinity)
            .padding(.vertical, 14)
            .background(OnboardingStyle.accent)
            .clipShape(RoundedRectangle(cornerRadius: OnboardingStyle.radiusButton))
        }
        .disabled(!enabled)
    }
}
